simtools/utils/general.py

```diff
@@ -975,3 +983,111 @@ def user_confirm():
         except EOFError:
             break
     return False
+
+
+def validate_data_type(reference_dtype, value=None, dtype=None, allow_subtypes=True):
+    """
+    Validate data type of value (scalar, list, np array) or type object against a
+    reference data type. Allow to check for exact data type or allow sub types
+    (e.g. uint is accepted for int).  Take into account 'file' type as used in the
+    model parameter database
+
+    Parameters
+    ----------
+    reference_dtype: str
+        Reference data type to be checked against.
+    value: any
+        Value to be checked (if dtype is None).
+    dtype: type
+        Type object to be checked (if value is None).
+    allow_subtypes: bool
+        If True, allow subtypes to be accepted.
+
+    Returns
+    -------
+    bool:
+        True if the data type is valid.
+
+    """
+    if value is not None and dtype is None:
+        if isinstance(value, (list, np.ndarray)):
+            value = np.array(value)
+            dtype = value.dtype
+        else:
+            dtype = type(value)
+    elif value is None and dtype is None:
+        raise ValueError("Either value or dtype must be given.")
+
+    # strict comparison
+    if not allow_subtypes:
+        return np.issubdtype(dtype, reference_dtype)
+    # allow any sub-type of integer or float for success
+    # dtype is 'object' for 'file' type and value None
+    if (np.issubdtype(dtype, np.str_) or np.issubdtype(dtype, "object")) and reference_dtype in (
+        "string",
+        "str",
+        "file",
+    ):
+        return True
+    if np.issubdtype(dtype, np.bool_) and reference_dtype in ("boolean", "bool"):
+        return True
+    # allow ints to be converted to floats
+    if np.issubdtype(dtype, np.integer) and (
+        np.issubdtype(reference_dtype, np.integer) or np.issubdtype(reference_dtype, np.floating)
+    ):
+        return True
+    if np.issubdtype(dtype, np.floating) and np.issubdtype(reference_dtype, np.floating):
+        return True
+
+    return False
+
+
+def convert_list_to_string(data, comma_separated=False):
```

Comment:
Tried to find a better name here, as it is a more general function that takes arrays as well. Maybe there is a category to describe both?

Reply:
I see what you mean. The nice thing now is that it clearly indicates that this is the 'mirror function' to `convert_string_to_list` (which really converts to a string).

Would you prefer `convert_list_or_ndarray_to_string()? Not sure if this really improves readability, and have no other idea.

Reply:
Ok I also don't have a better idea. Let's keep it this way.